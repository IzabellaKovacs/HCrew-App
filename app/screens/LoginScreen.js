import React from 'react';
import {StyleSheet, View, Image, Text, TouchableOpacity, TextInput,Alert} from 'react-native';

import firebase from '../../ApiKeys';

import * as Facebook from 'expo-facebook';

export default class LoginScreen extends React.Component{

    constructor(props){
        super(props)

        this.state=({
            email:'',
            password:''
        })
    }

    loginUser = (email,password) => {

        if(email == "") {
            alert("Write your email!");
        } else if(password == "") {
            alert("Write your password!");
        } else {
            firebase.auth().signInWithEmailAndPassword(email,password).then( () => {
               this.props.navigation.navigate('StartScreen');
            })
             .catch( function(error) {
                var errorCode = error.code;
                if(errorCode == 'auth/invalid-email')
                alert("Invalid Email!");
                else if(errorCode == 'auth/user-not-found') 
                alert("User not found!");
                else if(errorCode == 'auth/wrong-password') 
                alert("Wrong Password!"); 
            });
        }
    }
    

    async loginWithFacebook() {

          await Facebook.initializeAsync({appId: '<APP_ID>'});
          const {type,token} = await Facebook.logInWithReadPermissionsAsync({permissions: ['public_profile']});

          if (type === 'success') {

            const response = await fetch(`https://graph.facebook.com/me?access_token=${token}`);
            Alert.alert('Logged in!', `Hi ${(await response.json()).name}!`);

            const credential = firebase.auth.FacebookAuthProvider.credential(token);
            firebase.auth().signInWithCredential(credential).catch(error => {console.log(error)});
            this.props.navigation.navigate('StartScreen');
            
          }
       
      }

    render(){
        return(
        <View style = {styles.container}>
            <Image style={styles.logo} source={require('../assets/lgrg.png')}/>

            <TextInput style={{top: -50, height: 50, width: 300, justifyContent: 'center', backgroundColor: 'white', borderRadius: 15}}
                       placeholder=" Email" 
                       placeholderTextColor="#771D98"
                       autoCapitalize="none"
                       autoCorrect={false}
                       onChangeText={(email)=>this.setState({email})}
            />
            <TextInput style={{top: -40, height: 50, width: 300, justifyContent: 'center', backgroundColor: 'white', borderRadius: 15}}
                       placeholder=" Password" 
                       placeholderTextColor="#771D98"
                       secureTextEntry={true}
                       onChangeText={(password)=>this.setState({password})}
            />
            <TouchableOpacity onPress={() => this.loginUser(this.state.email,this.state.password)} style={styles.buttonContainer}>
                 <Text style={{textAlign: 'center', fontSize: 28, top: 0, color: 'white', fontWeight: 'bold'}}>Log in</Text>
            </TouchableOpacity>

            <Text style={{top: 5, color:"#771D98", fontWeight:'bold',fontSize:20}}>or with</Text>

            <View style={{flexDirection:'row'}} >
                <TouchableOpacity onPress={() => this.loginWithFacebook()}>
                <Image style={{width:50,height:50,position:'absolute',top:10, marginHorizontal:10}} source={require('../assets/fb.png')}/>
                </TouchableOpacity>
                <TouchableOpacity>
                <Image style={{width:50,height:50,position:'absolute',top:10,marginHorizontal:-50}} source={require('../assets/gm.png')}/>
                </TouchableOpacity>
            </View>

            
        </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#E7CDEA',
        alignItems: 'center',
        justifyContent: 'center',
    },
    logo: {
        width: 200, 
        height: 200,
        position: 'absolute',
        top: 60,
    },
    buttonContainer: {
        top: -10,
        height: 50,
        width: 300,
        justifyContent: 'center',
        backgroundColor: '#771D98',
        borderRadius: 25,
     }
});

