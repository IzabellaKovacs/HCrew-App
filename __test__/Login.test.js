import React from 'react';
import LoginScreen from '../app/screens/LoginScreen';

import { create } from 'react-test-renderer';
import renderer from 'react-test-renderer';

const tree = create(<LoginScreen />);

test('snapshot renders correctly', () => {
    expect(tree).toMatchSnapshot();
});

it('login: empty email test', () => {
    let data = renderer.create(<LoginScreen />).getInstance();

    window.alert = jest.fn();
    data.loginUser('','');
    
    expect(window.alert).toHaveBeenCalledTimes(1);
    expect(window.alert).toHaveBeenCalledWith("Write your email!");
})

it('login: empty password test', () => {
    let data = renderer.create(<LoginScreen />).getInstance();

    window.alert = jest.fn();
    data.setState({email: 'username@gmail.com'});
    data.loginUser(data.state.email,'');
    
    expect(window.alert).toHaveBeenCalledTimes(1);
    expect(window.alert).toHaveBeenCalledWith("Write your password!");
})