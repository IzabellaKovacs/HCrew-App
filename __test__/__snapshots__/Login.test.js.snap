// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E7CDEA",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 200,
        "position": "absolute",
        "top": 60,
        "width": 200,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    onChangeText={[Function]}
    placeholder=" Email"
    placeholderTextColor="#771D98"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 15,
        "height": 50,
        "justifyContent": "center",
        "top": -50,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder=" Password"
    placeholderTextColor="#771D98"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 15,
        "height": 50,
        "justifyContent": "center",
        "top": -40,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#771D98",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "top": -10,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 28,
          "fontWeight": "bold",
          "textAlign": "center",
          "top": 0,
        }
      }
    >
      Log in
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#771D98",
        "fontSize": 20,
        "fontWeight": "bold",
        "top": 5,
      }
    }
  >
    or with
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "marginHorizontal": 10,
            "position": "absolute",
            "top": 10,
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "marginHorizontal": -50,
            "position": "absolute",
            "top": 10,
            "width": 50,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E7CDEA",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 200,
        "position": "absolute",
        "top": 60,
        "width": 200,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    onChangeText={[Function]}
    placeholder=" Email"
    placeholderTextColor="#771D98"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 15,
        "height": 50,
        "justifyContent": "center",
        "top": -50,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder=" Password"
    placeholderTextColor="#771D98"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 15,
        "height": 50,
        "justifyContent": "center",
        "top": -40,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#771D98",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "top": -10,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 28,
          "fontWeight": "bold",
          "textAlign": "center",
          "top": 0,
        }
      }
    >
      Log in
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#771D98",
        "fontSize": 20,
        "fontWeight": "bold",
        "top": 5,
      }
    }
  >
    or with
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "marginHorizontal": 10,
            "position": "absolute",
            "top": 10,
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "marginHorizontal": -50,
            "position": "absolute",
            "top": 10,
            "width": 50,
          }
        }
      />
    </View>
  </View>
</View>
`;
